CHANGELOG.md

```diff
@@ -1,5 +1,31 @@
 # Optimizely C# SDK Changelog
 
+## 4.0.0
+January 16th, 2024
+
+### New Features
+- Promote Advanced Audience Targeting to General Availability
```

Comment:
I think this is fine with CHANGELOG, but we need the full description (copy from 4.0.0-beta) for ODP feature in release note.